src/go/rpk/pkg/cli/debug/bundle/bundle_k8s_linux.go

```diff
@@ -387,32 +388,75 @@ func saveSingleAdminAPICalls(ctx context.Context, ps *stepParams, fs afero.Fs, p
 					}
 					return requestAndSave(ctx, ps, fmt.Sprintf("admin/cpu_profile_%v.json", aName), f)
 				},
-				func() error {
-					err := requestAndSave(ctx, ps, fmt.Sprintf("metrics/%v/t0_metrics.txt", aName), cl.PrometheusMetrics)
-					if err != nil {
-						return err
-					}
-					select {
-					case <-time.After(metricsInterval):
-						return requestAndSave(ctx, ps, fmt.Sprintf("metrics/%v/t1_metrics.txt", aName), cl.PrometheusMetrics)
-					case <-ctx.Done():
-						return requestAndSave(ctx, ps, fmt.Sprintf("metrics/%v/t1_metrics.txt", aName), cl.PrometheusMetrics)
-					}
+			}
+			funcs = append(funcs, r...)
+		}
+
+		var grp multierror.Group
+		for _, f := range funcs {
+			grp.Go(f)
+		}
+		errs := grp.Wait()
+		if errs != nil {
+			rerrs = multierror.Append(rerrs, errs)
+		}
+		return rerrs.ErrorOrNil()
+	}
+}
+
+func saveMetricsAPICalls(ctx context.Context, ps *stepParams, fs afero.Fs, p *config.RpkProfile, adminAddresses []string, metricsInterval time.Duration, metricsSampleCount int) step {
+	return func() error {
+		var rerrs *multierror.Error
+		var funcs []func() error
+		for _, a := range adminAddresses {
+			a := a
+			p = &config.RpkProfile{
+				KafkaAPI: config.RpkKafkaAPI{
+					SASL: p.KafkaAPI.SASL,
 				},
-				func() error {
-					err := requestAndSave(ctx, ps, fmt.Sprintf("metrics/%v/t0_public_metrics.txt", aName), cl.PublicMetrics)
+				AdminAPI: config.RpkAdminAPI{
+					Addresses: []string{a},
+					TLS:       p.AdminAPI.TLS,
+				},
+			}
+			cl, err := adminapi.NewClient(fs, p)
+			if err != nil {
+				rerrs = multierror.Append(rerrs, fmt.Errorf("unable to initialize admin client for %q: %v", a, err))
+				continue
+			}
+
+			endpoints := map[string]func(context.Context) ([]byte, error){"metrics": cl.PrometheusMetrics, "public_metrics": cl.PublicMetrics}
+			aName := sanitizeName(a)
+			for endpointName, endpoint := range endpoints {
+				endpointPoller := func() error {
+					err := requestAndSave(ctx, ps, fmt.Sprintf("metrics/%v/t0_%s.txt", aName, endpointName), endpoint)
 					if err != nil {
 						return err
 					}
-					select {
-					case <-time.After(metricsInterval):
-						return requestAndSave(ctx, ps, fmt.Sprintf("metrics/%v/t1_public_metrics.txt", aName), cl.PublicMetrics)
-					case <-ctx.Done():
-						return requestAndSave(ctx, ps, fmt.Sprintf("metrics/%v/t1_public_metrics.txt", aName), cl.PublicMetrics)
+					ticker := time.NewTicker(metricsInterval)
+					defer ticker.Stop()
+
+					for count := range metricsSampleCount {
+						select {
+						case <-ticker.C:
+							count++
```

Comment:
Why do we need to increment the count if we are ranging over it?

Same question below

Reply:
Good point - although it's harmless as the iteration doesn't carry over to the range itself and it solves the problem of the value being off by one (because on the `0`th iteration, we're actually doing `t1_metrics.txt`), that's very messy. Will change.